### PR TITLE
Deactivation only for singleton

### LINKED ISSUE #1334: Deactivation errors should apply to request scope too
https://github.com/inversify/InversifyJS/blob/9f6f43e53752679a610684baceea98533ad65c4b/src/resolution/instantiation.ts#L119

Change to 
```
function _validateInstanceResolution(binding: interfaces.Binding<unknown>, constr: interfaces.Newable<unknown>): void {
   const scopeMessage =  `Class cannot be instantiated in ${binding.scope} scope.  Deactivation is for singletons.`
   if (binding.scope !== "Singleton") {
        if (typeof binding.onDeactivation === "function") {
            throw new Error(ON_DEACTIVATION_ERROR(constr.name, scopeMessage));
        }

        if (Reflect.hasMetadata(METADATA_KEY.PRE_DESTROY, constr)) {
            throw new Error(PRE_DESTROY_ERROR(constr.name, scopeMessage));
        }
    }
}

```

### PR DESCRIPTION
**Pull request by tonyhallett**
fix #1334 

I did not suffix the error with 'Deactivation is for singletons.'  Add the suffix ?

### CODE REVIEW

File: src/resolution/instantiation.ts [hunk 1]
```diff
@@ -116,13 +116,15 @@ function _postConstruct<T>(constr: interfaces.Newable<T>, instance: T): void | P
...
         if (typeof binding.onDeactivation === "function") {
-            throw new Error(ON_DEACTIVATION_ERROR(constr.name, "Class cannot be instantiated in transient scope."));
+            throw new Error(ON_DEACTIVATION_ERROR(constr.name, scopeErrorMessage));
         }
 
         if (Reflect.hasMetadata(METADATA_KEY.PRE_DESTROY, constr)) {
-            throw new Error(PRE_DESTROY_ERROR(constr.name, "Class cannot be instantiated in transient scope."));
+            throw new Error(PRE_DESTROY_ERROR(constr.name, scopeErrorMessage));
         }
     }
```
Comment (notaphplover) [inline src/resolution/instantiation.ts]: We should use `BindingTypeEnum` instead

**Review by notaphplover — CHANGES_REQUESTED** [review]
Just a minor change is requested

File: src/resolution/instantiation.ts (hunk 1 above)
Comment (tonyhallett) [reply src/resolution/instantiation.ts]: Ok

File: test/resolution/resolver.test.ts [hunk 2]
```diff
@@ -1475,6 +1475,37 @@ describe("Resolve", () => {
           .throw("onDeactivation() error in class Destroyable: Class cannot be instantiated in transient scope.");
   });
 
+  it("Should not allow request construction with preDestroy", async () => {
+      @injectable()
+      class Destroyable {
+          @preDestroy()
+          public myPreDestroyMethod() {
+              //
```
Comment (notaphplover) [inline test/resolution/resolver.test.ts]: if you want to remove the empty body warning, you can return `undefined` instead

**Review by notaphplover — APPROVED** [review]
LGTM!

**Pull request merged**